marquez/client.py

```diff
@@ -0,0 +1,237 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import collections
+import os
+import urllib
+
+import requests
+
+from marquez.constants import JobRunState
+from marquez.utils import (InvalidRequestException,
+                           handle_response, _compose_path)
+
+
+class MarquezClient(object):
+    API_PATH = "api/v1"
+    DEFAULT_TIMEOUT_MILLISECONDS = 5000
+    marquez_host = None
+    namespace = None
+
+    def __init__(self, host=None, port=None, timeout=None):
+        host = os.environ.get('MARQUEZ_HOST') or host
+        port = os.environ.get('MARQUEZ_PORT') or (
+            str(port) if port else port)
+        t = os.environ.get('MARQUEZ_TIMEOUT_MS') or str(timeout)
+        timeout_milliseconds = (int(t) if t.isnumeric() else
+                                self.DEFAULT_TIMEOUT_MILLISECONDS)
+        self.timeout = float(timeout_milliseconds / 1000)
+
+        if not host or not port:
+            msg = ("Please provide host & port or set the proper env "
+                   "vars: MARQUEZ_HOST, MARQUEZ_PORT")
+            raise ConnectionError(msg)
+
+        self.marquez_host = "{0}:{1}/{2}".format(
+            host, port, self.API_PATH)
+
+        self._namespace = None
+
+    @property
+    def namespace(self):
+        if not self._namespace:
+            raise InvalidRequestException("Please set a namespace")
+        return self._namespace
+
+    def set_namespace(self, ns_name, owner=None, description=None):
+        self.create_namespace(ns_name, owner or 'default', description)
+        self._namespace = urllib.parse.quote_plus(ns_name)
+
+    #  Namespaces API
+    @handle_response
+    def get_namespace(self, ns_name, **kwargs):
```

Comment:
_Minor_: Mind spelling out `ns_name` as `namespace`. The abbreviation doesn't provide much in readability and saves only a few characters.

Reply:
Fixed